File: Merge 2 sorted linked list in reverse order.cpp
class Solution
{
    public:
    
    Node* Reverse(Node* node)
    {
        if(node==NULL)
        {
            return node;
        }
        if(node->next==NULL)
        {
            return node;
        }
        Node* cur=node;
        Node* prev=NULL;
        Node* nex=node->next;
        while(nex!=NULL)
        {
            cur->next=prev;
            prev=cur;
            cur=nex;
            nex=nex->next;
        }
        cur->next=prev;
        return cur;
    }
     Node* Merge(Node* n1, Node* n2)
    {
        Node* one=n1;
        Node* two=n2;
        Node* prev;
        Node* ans;
        if(n1->data > n2->data)
        {
            ans=n1;
            prev=n1;
            n1=n1->next;
        }
        else
        {
            Node* temp=n1;
            n1=n2;
            n2=temp;
            ans=n1;
            prev=n1;
            n1=n1->next;
        }
        while(n1!=NULL && n2!=NULL)
        {
            if(n1->data>=n2->data)
            {
                prev=n1;
                n1=n1->next;
            }
            else
            {
                prev->next=n2;
                prev=n2;
                Node* temp=n1;
                n1=n2->next;
                n2=temp;
            }
        }
        if(n2!=NULL)
        {
           prev->next =n2;
        }
        return ans;
    }
    struct Node * mergeResult(Node *node1,Node *node2)
    {
        Node* n1=Reverse(node1);
        Node* n2=Reverse(node2);
        if(n1!=NULL && n2!=NULL)
        {Node* node=Merge(n1,n2);
        return node;}
        else
        {
            if(n1==NULL)
            return n2;
            
            if(n2==NULL)
            return n1;
        }
    }  
};

File: Merge 2 sorted linked list in reverse order.java
class GfG
{
    Node mergeResult(Node node1, Node node2)
    {
        
        Node temp1 = node1;
        Node temp2 = node2;
        
        ArrayList<Integer> array = new ArrayList<Integer>();
        
        while(temp1!=null){
            array.add(temp1.data);
            temp1 = temp1.next;
        }
        
        while(temp2!=null){
            array.add(temp2.data);
            temp2 = temp2.next;
        }
        
        Collections.sort(array);
        
        Node head = null;
        Node tail = null;
        
        for(int i=(array.size()-1);i>=0;i--){
            Node newNode = new Node(array.get(i));
            
            if(head==null){
                head = tail = newNode;
            }
            else{
                while(tail.next!=null){
                    tail = tail.next;
                }
                tail.next = newNode;
                tail = newNode;
            }
        }
        
        return head;
    }
}
